### PR TITLE
Questionnaire more devices

### PR DESCRIPTION
**Pull request by jsheppard95**
## Description:
Gave happi questionnaire backend access to analog in/out and EVR control trigger PV's if entered in questionnaire, making this backend include any device in the questionnaire that is given a PV.
Does not include "Digitizers: Diodes et al." which are not given PV names

## Testing:
`import happi.backends.qs_db`
`qs_backend = happi.backends.qs_db.QSBackend('mfxlu3417')`
`print(qs_backend.db)`
` # Returns dictionary with entered PV's in quetsionnaire: https://pswww.slac.stanford.edu/questionnaire_slac/proposal_questionnaire/run17/LU34/#pcdssetup`

### CODE REVIEW

File: happi/containers.py [hunk 1]
```diff
@@ -297,3 +297,43 @@ class Motor(Device):
...
+    system = copy(Device.system)
+    system.default = 'acromag'
+
+
+class ControlsTriggers(Device):
+    """
+    A Generic class for Controls Triggers
+    """
+    device_class = copy(Device.device_class)
+    device_class.default = 'pcdsdevices.device_types.ControlsTriggers'
```
Comment (teddyrendahl) [inline happi/containers.py]: These have to actually exist in the `pcdsdevices` module. I think this is called `Trigger` or something?

File: happi/containers.py [hunk 2]
```diff
@@ -297,3 +297,43 @@ class Motor(Device):
...
     system = copy(Device.system)
     system.default = 'motion'
+
+
+class Camera(Device):
+    """
+    A Generic EpicsCamera
+    """
+    device_class = copy(Device.device_class)
+    device_class.default = 'pcdsdevices.device_types.Camera'
```
Comment (teddyrendahl) [inline happi/containers.py]: What is the difference between `Camera` and `AreaDetector`? If the `Camera` objects are in the DAQ, let's ignore them for now.

File: happi/backends/qs_db.py [hunk 3]
```diff
@@ -81,13 +83,19 @@ def __init__(self, expname, **kwargs):
         raw = self.qs.getProposalDetailsForRun(run_no, proposal)
         for table, _class in self.device_translations.items():
             # Create a regex pattern to find all the appropriate pattern match
+            # example pattern for non-AO devices:
+            # 'pcdssetup-camera-setup-1-purpose'
+            # example pattern for AO devices:
+            # 'pcdssetup-ao-1-purpose'
             pattern = re.compile(r'pcdssetup-{}-'
                                  r'setup-(\d+)-(\w+)'.format(table))
+            pattern_analog = re.compile(r'pcdssetup-{}-'
```
Comment (teddyrendahl) [inline happi/backends/qs_db.py]: Let's wait and get this patched on the Questionnaire side.

**Comment by jsheppard95** [comment]
> It should be pretty easy to add a test here. There is a big JSON dump example from the Questionnaire, just add an example of each of these devices there.

So I realized while trying to figure out this test that because the cameras do not have PVs in the questionnaire they will not show up when using the client with a questionnaire backend.
Example:
In:
`import happi`
`import happi.backends.qs_db`
`qs_backend = happi.backends.qs_db.QSBackend('mfxlu3417')`
`client = happi.Client(qs_backend)`
`client.all_devices`
Out:
`[Motor trans (prefix=MFX:USR:MMN:41, z=-1.0),`
 `Motor laser_opo_waveplate (prefix=MFX:USR:MMN:09, z=-1.0),`
 `Motor laser_evo_waveplate (prefix=MFX:USR:MMN:11, z=-1.0),`
 `Motor laser_opo_x (prefix=MFX:LAS:PIC:01, z=-1.0),`
 `Motor laser_opo_y (prefix=MFX:LAS:PIC:02, z=-1.0),`
 `Motor sleeveY (prefix=MFX:USR:MMN:42, z=-1.0),`
 `Motor sleeveZ (prefix=MFX:USR:MMN:43, z=-1.0),`
 `Motor jetY (prefix=MFX:USR:MMN:44, z=-1.0),`
 `Motor jetZ (prefix=MFX:USR:MMN:47, z=-1.0),`
 `Motor tilt (prefix=MFX:USR:MMN:48, z=-1.0),`
 `Motor tapeX (prefix=MFX:USR:LI61:07, z=-1.0),`
 `Motor epixY (prefix=MFX:USR:LI61:08, z=-1.0),`
 `Motor nav_focus (prefix=MFX:USR:MMS:18, z=-1.0),`
 `Motor nav_zoom (prefix=MFX:USR:MMS:17, z=-1.0),`
 `Motor Inj X (prefix=MFX:USR:MMN:25, z=-1.0),`
 `Motor Inj Y (prefix=MFX:USR:MMN:26, z=-1.0),`
 `Trigger Overview_trig (prefix=MFX:REC:EVR:02:TRIG1, z=-1.0),`
 `Trigger Meniscus_trig (prefix=MFX:REC:EVR:02:TRIG3, z=-1.0),`
 `Trigger Inline_trig (prefix=MFX:REC:EVR:02:TRIG8, z=-1.0),`
 `Trigger Deposition_trig (prefix=MFX:REC:EVR:02:TRIG9, z=-1.0),`
 `Trigger Side_trig (prefix=MFX:REC:EVR:02:TRIGA, z=-1.0),`
 `Trigger XrayStim_trig (prefix=MFX:LAS:EVR:01:TRIG7, z=-1.0),`
 `Trigger Pacemaker_trig (prefix=MFX:LAS:EVR:01:TRIG4, z=-1.0),`
 `Trigger EVOQ_trig (prefix=MFX:LAS:EVR:01:TRIG5, z=-1.0),`
 `Trigger Inhibit_trig (prefix=MFX:LAS:EVR:01:TRIG6, z=-1.0)]`

The analog outputs will show once the questionnaire is patched, but the cameras won't. Is this something I should try to patch in the Client functions?

File: happi/containers.py (hunk 2 above)
Comment (jsheppard95) [reply happi/containers.py]: Removed reading cameras from questionnaire due to them not having PV's, will add back later once questionnaire is patched

**Comment by jsheppard95** [comment]
I messed up the history for this branch in an attempt to resolve a merge conflict. It looks okay now since I force pushed a commit before the merge conflict, but having trouble getting my local branch to agree with the remote. I'll instead close this PR, delete my remote branch, and create a new remote branch/PR that agrees with my local version which has the correct history

**Pull request closed without merging**